app/views/projects/samples/attachments/_attachment.html.erb

```diff
@@ -2,18 +2,82 @@
   id="<%= dom_id(attachment) %>"
   class="bg-white border-b dark:bg-slate-800 dark:border-slate-700"
 >
-  <th scope="row" class="px-6 py-4"><%= link_to attachment.file.filename,
-    download_namespace_project_sample_attachment_path(
-      sample_id: @sample.id,
-      id: attachment.id
-    ),
-    data: {
-      turbo: false
-    } %></th>
-  <td class="px-6 py-4"><%= number_to_human_size(attachment.file.blob.byte_size) %></td>
+  <% if attachment.associated_attachment && attachment.metadata['direction'] == 'forward' %>
+    <td class="px-6 py-4">
+      <div>
+        <div class="flex mb-4 items-center">
+          <%= viral_icon(name: :arrow_right, color: :subdued, classes: "h-6 w-6 ml-0 mr-2") %>
+          <span>
+            <%= link_to attachment.file.filename,
+            download_namespace_project_sample_attachment_path(
+              sample_id: @sample.id,
+              id: attachment.id
+            ),
+            data: {
+              turbo: false
+            } %>
+          </span>
+        </div>
+        <div class="flex items-center">
+          <%= viral_icon(name: :arrow_left, color: :subdued, classes: "h-6 w-6 ml-0 mr-2") %>
+          <span>
+            <%= link_to attachment.associated_attachment_filename,
+            download_namespace_project_sample_attachment_path(
+              sample_id: @sample.id,
+              id: attachment.metadata["associated_attachment_id"]
+            ),
+            data: {
+              turbo: false
+            } %>
+          </span>
+        </div>
+      </div>
+    </td>
+  <% else %>
+    <td class="px-6 py-4"><%= link_to attachment.file.filename,
+      download_namespace_project_sample_attachment_path(
+        sample_id: @sample.id,
+        id: attachment.id
+      ),
+      data: {
+        turbo: false
+      } %>
+    </td>
+  <% end %>
+
+  <% if attachment.metadata['format'] == 'fasta' %>
+    <% format_color = "blue" %>
+  <% elsif attachment.metadata['format'] == 'fastq' %>
+    <% format_color = "green" %>
+  <% elsif attachment.metadata['format'] == 'unknown' %>
+    <% format_color = "gray" %>
+  <% end %>
+  <td class="px-6 py-4"><%= viral_pill(text: attachment.metadata["format"], color: format_color) %></td>
+
+  <% if attachment.metadata['type'] == 'assembly' %>
```

Comment:
Same comment as above about moving into a helper. You can put this in a separate method in the same helper file

Reply:
Same as above f93ae17f6874db9352897610bc2fa5ff92ec5f3a

Reply:
Fixed in 9fde28e1bab0cebc9ae7142848f7aa2f747db687